pandas/_libs/index_class_helper.pxi.in

```diff
@@ -12,16 +12,26 @@ WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
 
 # name, dtype, ctype
 dtypes = [('Float64', 'float64', 'float64_t'),
-          ('UInt64', 'uint64', 'uint64_t'),
+          ('Float32', 'float32', 'float32_t'),
           ('Int64', 'int64', 'int64_t'),
-          ('Object', 'object', 'object')]
+          ('Int32', 'int32', 'int32_t'),
+          ('Int16', 'int16', 'int16_t'),
+          ('Int8', 'int8', 'int8_t'),
+          ('UInt64', 'uint64', 'uint64_t'),
+          ('UInt32', 'uint32', 'uint32_t'),
+          ('UInt16', 'uint16', 'uint16_t'),
+          ('UInt8', 'uint8', 'uint8_t'),
```

Comment:
Are the lower Uint engines actually used somewhere in the code? (I think you only changed things in Categorical, and those only use the Int.. engines?)

So if they are not used, should we actually add them?

Reply:
The lower UInt engine are not used in the code ATM. I added them by suggestion from @jreback. I don't have a super strong preference on whether to add them now, but I can see benefits to aving them, e.g.:
* if someone want to make specialized indexes based on UInts they'd be nice to have.
* We don't only add UInt8/16/32 engines but also various UInt8/16/32 algorithms, that could be beneficial to have around for when the need is there.

all in all I'm weakly positive on adding them now, but will let you core devs decide on it.